datastructures/linkedlist/cyclicallylinkedlist/cyclicallylinkedlist_test.go

```diff
@@ -0,0 +1,108 @@
+package cyclicallylinkedlist
+
+import (
+	"reflect"
+	"testing"
+)
+
+type testCase struct {
+	param        int
+	wantToReturn int
+	listCount    int
+}
+
+func fillList(list *ClList, n int) {
+	for i := 1; i <= n; i++ {
+		list.Add(i)
+	}
+}
+
+// 100% coverage.
+func TestLinkedList(t *testing.T) {
+	list := NewList()
+	list.Add(1)
+	list.Add(2)
+	list.Add(3)
+
+	t.Run("Test Add()", func(t *testing.T) {
```

Comment:
You should separate out the tests I think. That way if someone writes a different implementation on top of your algorithms, they can test different implementations of the methods (such as `Delete`, `Add`, `Rotate` etc) without failing other tests of the things that they are not changing.

Reply:
Can you check my changes and resolve this if all is well?